index.js

```diff
@@ -4281,17 +4316,30 @@ Agent.prototype.request = function (requestMessage, rinfo) {
 						value: null
 					});
 				};
+			} else if ( ! this.isAllowed(requestPdu.type, providerNode.provider.maxAccess, providerNode.provider ) ) {
+					// requested access not allowed (by MAX-ACCESS)
+					mibRequests[i] = new MibRequest ({
+						operation: requestPdu.type,
+						oid: requestPdu.varbinds[i].oid
+					});
+					handlers[i] = function getRanaHandler (mibRequestForRana) {
+						mibRequestForRana.done ({
+							errorStatus: ErrorStatus.NoAccess,
+							errorIndex: 0,
+							type: ObjectType.Null,
+							value: null
+						});
+					};
 			} else {
 				mibRequests[i] = new MibRequest ({
 					operation: requestPdu.type,
 					providerNode: providerNode,
 					instanceNode: instanceNode,
 					oid: requestPdu.varbinds[i].oid
 				});
-				if ( requestPdu.type == PduType.SetRequest ) {
```

Comment:
What is the reason for removing the PDU SetRequest type check?  The `setType` and `setValue` only pertain to SetRequests, so leaving these on getRequests is meaningless/clutter at best, if not deceptive.  As such, I'm inclined to keep this check.

Reply:
I can't see any good reason to elide it. I expect I deleted it by accident. I'll re-insert that test.